engines/supernova/game-manager.cpp

```diff
@@ -0,0 +1,1022 @@
+		sentence(_currentSentence, false);
+		_currentSentence = a;
+		sentence(_currentSentence, true);
+	}
+}
+
+void GameManager::takeObject(Object &obj) {
+	if (obj.hasProperty(CARRIED))
+		return;
+
+	if (obj._section != 0)
+		_vm->renderImage(obj._section);
+	obj._click = obj._click2 = 255;
+	_inventory.add(obj);
+}
+
+void GameManager::drawCommandBox() {
+	for (int i = 0; i < ARRAYSIZE(_guiCommandButton); ++i) {
+		_vm->renderBox(_guiCommandButton[i]);
+		int space = (_guiCommandButton[i].width() - Screen::textWidth(_guiCommandButton[i].getText())) / 2;
+		_vm->renderText(_guiCommandButton[i].getText(),
+						_guiCommandButton[i].getTextPos().x + space,
+						_guiCommandButton[i].getTextPos().y,
+						_guiCommandButton[i].getTextColor());
+	}
+}
+
+void GameManager::drawInventory() {
+	for (int i = 0; i < ARRAYSIZE(_guiInventory); ++i) {
+		_vm->renderBox(_guiInventory[i]);
+		_vm->renderText(_inventory.get(i + _inventoryScroll)->_name,
+						_guiInventory[i].getTextPos().x,
+						_guiInventory[i].getTextPos().y,
+						_guiInventory[i].getTextColor());
+	}
+
+	_vm->renderBox(_guiInventoryArrow[0]);
+	_vm->renderBox(_guiInventoryArrow[1]);
+	if (_inventory.getSize() > ARRAYSIZE(_guiInventory)) {
+		if (_inventoryScroll != 0)
+			_vm->renderText(_guiInventoryArrow[0]);
+		if (_inventoryScroll + ARRAYSIZE(_guiInventory) < _inventory.getSize())
+			_vm->renderText(_guiInventoryArrow[1]);
+	}
+}
+
+int GameManager::getKeyInput() {
+	while (!_vm->shouldQuit()) {
+		updateEvents();
+		if (_keyPressed) {
+			return _key.ascii;
+		}
+		g_system->updateScreen();
+		g_system->delayMillis(_vm->_delay);
+	}
+	return 0;
+}
+
+Common::EventType GameManager::getMouseInput() {
+	while (!_vm->shouldQuit()) {
+		updateEvents();
+		if (_mouseClicked)
+			return _mouseClickType;
+		g_system->updateScreen();
+		g_system->delayMillis(_vm->_delay);
+	}
+	return Common::EVENT_INVALID;
+}
+
+void GameManager::getInput() {
+	while (!_vm->shouldQuit()) {
+		updateEvents();
+		if (_mouseClicked || _keyPressed)
+			break;
+		g_system->updateScreen();
+		g_system->delayMillis(_vm->_delay);
+	}
+}
+
+void GameManager::roomBrightness() {
+}
+
+void GameManager::changeRoom(RoomId id) {
+	_currentRoom = _rooms[id];
+	_newRoom = true;
+
+	for (int i = 0; i < 25; i++)
+		_exitList[i] = -1;
+	for (int i = 0; i < kMaxObject; i++) {
+		if (_currentRoom->getObject(i)->hasProperty(EXIT)) {
+			byte r = _currentRoom->getObject(i)->_direction;
+			_exitList[r] = i;
+		}
+	}
+}
+
+void GameManager::wait(int ticks) {
+	int32 end = _time + ticksToMsec(ticks);
+	do {
+		g_system->delayMillis(_vm->_delay);
+		updateEvents();
+		g_system->updateScreen();
+	} while (_time < end && !_vm->shouldQuit());
+}
+
+void GameManager::waitOnInput(int ticks) {
+	int32 end = _time + ticksToMsec(ticks);
+	do {
+		g_system->delayMillis(_vm->_delay);
+		updateEvents();
+		g_system->updateScreen();
+	} while (_time < end && !_vm->shouldQuit() && !_keyPressed && !_mouseClicked);
+}
```

Comment:
The input check could be a separate input parameter flag, so these two could be merged